Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -783,14 +799,14 @@ public final class ManifestLoader: ManifestLoaderProtocol {
             let diagnosticFile = diaDir.appending(component: "\(packageIdentity).dia")
             try localFileSystem.createDirectory(diaDir, recursive: true)
             cmd += ["-Xfrontend", "-serialize-diagnostics-path", "-Xfrontend", diagnosticFile.pathString]
-            result.diagnosticFile = diagnosticFile
+            evaluationResult.diagnosticFile = diagnosticFile
         }
 
         cmd += [manifestPath.pathString]
 
         cmd += self.extraManifestFlags
 
-        try withTemporaryDirectory(removeTreeOnDeinit: true) { tmpDir in
+        try withTemporaryDirectory { tmpDir, cleanupTmpDir in
```

Comment:
how about we change `withTemporaryDirectory` to take a  "cleanWhenDone" argument instead of have the "cleanup" argument passed to the closure? feels nicer?

Reply:
Sounds like another PR to me.